pyradiosky/utils.py

```diff
@@ -278,3 +280,113 @@ def download_gleam(path=".", filename="gleam.vot", overwrite=False, row_limit=No
     table.write(opath, format="votable", overwrite=overwrite)
 
     print("GLEAM catalog downloaded and saved to " + opath)
+
+
+def modified_gleam(
+    gleam_filename="gleamegc.dat",
+    usecols=(10, 12, 77, -5),
+    fill_blank=False,
+    nside=32,
+    add_peeled_sources=False,
+    modified_gleam_filename="",
+):
+    """
+    Write modified gleam catalogue to fill the blank regions, and also to add the peeled sources.
+
+    Parameters
+    ----------
+    gleam_filename : str
+        name of the gleam catalogue file.
+    usecols : tuple
+        default = (10,12,77,-5)
+        4 columns, 1st-RA (deg), 2nd-Dec (deg), 3rd - Flux at 100 MHz (Jy), 4th - sp. index
+    fill_blank : bool
+        If True, it will fill the blank regions
+    nside : int
+        Use for healpix pixalization
+    add_peeled_sources : bool
+        If True, it will add the peeled sources from Table 2 Hurley-Walker, N et al. 2017
+    modified_gleam_filename : str
+        name of the modified gleam catalogue file.
+
+    """
+    import astropy_healpix as astrohp
+    import astropy_healpix.healpy as hp
+
+    # check if file exists in the given path
+    if os.path.exists(gleam_filename):
+        print("true gleam catalogue file - ", gleam_filename)
+    else:
+        raise ValueError("File not found")
+
+    # read array of RA (deg), Dec (deg), Flux at 100 MHz (Jy), sp. index
+    aa = np.genfromtxt(gleam_filename, usecols=usecols)
```

Comment:
Is there a good reason not to use the VOT table and the `read_gleam_catalog` method on SkyModel to get the values you need here?